doc/user/data/sql_funcs.yml

```diff
@@ -207,6 +207,9 @@
   - signature: 'char_length(s: str) -> int'
     description: Number of code points in `s`
 
+  - signature: 'convert_from(b: bytea, src_encoding: text) -> text'
+    description: Convert data `b` from original encoding specified by `src_encoding` into text with current database encoding.
```

Comment:
I may be misunderstanding convert_from, but I don't think the signature is correct. `b: bytea` would mean that `b` *has* to be encoded in bytea format, which is different from what the description says, and also different from the example in "Create Source".  In the "Create Source" example, `data` is a column name, not a description, and the second term is actually the format you're coverting *to*, not from. The list of permitted psql encodings also doesn't include `text`; I think you want `SQL_ASCII` or `UTF8`.

I think the correct signature would be `'convert_from(col_name, new_encoding) -> new_encoding'` and the correct description would be "Convert column `col_name` to `new_encoding`. `New_encoding` can be any character set supported by [Postgres](https://www.postgresql.org/docs/9.5/multibyte.html)."

But you should definitely have this reviewed by an engineer for technical correctness first.

Reply:
> I may be misunderstanding convert_from, but I don't think the signature is correct. `b: bytea` would mean that `b` _has_ to be encoded in bytea format, which is different from what the description says, and also different from the example in "Create Source".

I think you're actually misunderstanding how the function signatures work here. `: bytea` here indicates only that the expression passed as the first parameter must have type `bytea`. Many expressions have type `bytea`:

  * `data`, where `data` is a column of type bytea
  * A byte literal like '\x12'::bytea
  * A more complicated construction of a bytea value: `'\x12'::bytea || '\x23'`

In the case of `convert_from`, the first argument does indeed have type `bytea`. The encoding of those bytes is not part of the type information, which is why you're required to specify which encoding is in use via the `src_encoding` parameter.

> The list of permitted psql encodings also doesn't include `text`; I think you want `SQL_ASCII` or `UTF8`...

Similar to above; `text` here is the name of a type, not the name of an encoding.

> ...the second term is actually the format you're coverting to, not from

No, it is in fact the format you're converting from. It just so happens that in that `CREATE SOURCE` example you are both converting _from_ and _to_ UTF-8.